docs/firmware/golioth-firmware-sdk/light-db-stream/guide-lightdb-stream-object.md

```diff
@@ -0,0 +1,235 @@
+---
+id: by-object
+title: LightDB Stream (by object)
+sidebar_position: 2
+---
+
+Object-based LightDB functions can be used to send any number of time-series
+data values to the cloud with a single function call. A timestamp is
+automatically added to the data by the server and stored using a specific path
+that you provide. This data will accumulate along with all previously received
+values. If the chosen path does not already exist it will be created.
+
+Synchronous and asynchronous functions exist to set multiple data types and
+paths using your choice of these object types: `JSON` or `CBOR`.
+
+## Includes
+
+```c
+#include "golioth.h"
+```
+
+Including the `golioth.h` header file makes the Golioth API functions available
+to your program.
+
+## Preparing the Data
+
+The data you want to send must first be prepared as an object. This will include
+one or more key-values pairs. Choose between `JSON` or `CBOR` format.
+
+### JSON
+
+JSON payloads may be formatted using simple `snprintf()` style string
+substitution or a library like [cJSON](https://github.com/DaveGamble/cJSON).
+Here is a properly-formatted JSON example using all of the key-value pair
+examples from [the type-based stream examples](by-type#synchronous):
+
+```c
+/* Create a valid JSON string */
+char json_buf[] = "{\"name\":\"Golioth\",\"active\":true,\"data\":{\"reading\":1.337,\"calib\":42}}";
```

Comment:
It would be good to include a description of this object somewhere on this page - what are all the possible fields, etc. I'm not sure if we already have a schema defined somewhere for this but that could be useful as well.